hydra-cluster/src/CardanoNode.hs

```diff
@@ -125,6 +125,17 @@ getCardanoNodeVersion :: IO String
 getCardanoNodeVersion =
   readProcess "cardano-node" ["--version"] ""
 
+-- | Tries to reach for an existing cardano-node running on default state directory.
+findRunningCardanoNode :: FilePath -> KnownNetwork -> IO (Maybe RunningNode)
+findRunningCardanoNode stateDirectory knownNetwork = do
+  let knownNetworkId = toNetworId knownNetwork
+  n@RunningNode{networkId, nodeSocket} <- mkRunningNode stateDirectory defaultCardanoNodeArgs
+  guard (knownNetworkId == networkId)
+  try (queryTip knownNetworkId nodeSocket)
+    >>= \case
+      Right _ -> pure Nothing
+      Left (_ :: SomeException) -> do pure (Just n)
```

Comment:
Must: flip these two things. 

The `Right` case is that we successfully can use this node (with this network id). So in that case we want to return `Just n`, where i n the `Left` we have a problem to talk to this node and we should return `Nothing`

Reply:
my bad